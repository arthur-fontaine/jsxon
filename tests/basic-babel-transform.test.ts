import { test, describe, expect } from "vitest";
import * as babel from "@babel/core";
import transformPlugin from "@babel/plugin-transform-react-jsx";

describe("basic babel transform", () => {
	test("should transform jsx", () => {
		const code = /* tsx */ `
      import JSXon, { Text } from '../src/index.ts'
      export const element = <Text text="Hello, JSXon!" />
    `;
		const result = babel.transformSync(code, {
			plugins: [
				[
					transformPlugin,
					{
						pragma: "JSXon.createElement",
						pragmaFrag: "JSXon.Fragment",
					},
				],
			],
		});

		const transformedCode = result?.code;
		expect(transformedCode).toMatchSnapshot();
	});
});
