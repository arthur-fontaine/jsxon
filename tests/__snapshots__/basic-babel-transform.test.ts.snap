// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic babel transform > should transform jsx 1`] = `
"import JSXon, { Text } from '../src/index.ts';
export const element = JSXon.createElement(Text, {
  text: "Hello, JSXon!"
});"
`;
